src/lib/seam/components/AccessCodeForm/AccessCodeForm.tsx

```diff
@@ -0,0 +1,192 @@
+  const [name, setName] = useState('')
+  const [code, setCode] = useState('')
+  const [codeInputFocused, toggleCodeInputFocused] = useToggle()
+
+  const createAccessCode = useCreateAccessCode(device)
+
+  const save = (): void => {
+    if (name === '' || code === '') {
+      return
+    }
+
+    if (createAccessCode.isLoading) {
+      return
+    }
+
+    createAccessCode.mutate(
+      {
+        name,
+        code,
+      },
+      {
+        onSuccess: () => {
+          onBack?.()
+        },
+      }
+    )
+  }
+
+  const nameError = name.length > 60 ? t.overCharacterLimitError : undefined
+  const codeError = getCodeError(code)
+
+  const generateCode = (): void => {
+    // Randomize code length
+    const length = getRandomInt({ min: minCodeLength, max: maxCodeLength })
+    // Fill each digit with a random int from 0-9
+    const generated = Array.from({ length }, () =>
+      getRandomInt({ min: 0, max: 9 })
+    ).join('')
+
+    setCode(generated)
+  }
+
+  const isFormValid =
+    name.trim().length > 0 &&
+    nameError === undefined &&
+    code.trim().length > 0 &&
+    codeError === undefined &&
+    !createAccessCode.isLoading
+
+  return (
+    <>
+      <ContentHeader
+        title={t.addNewAccessCode}
+        subheading={device.properties.name}
+        onBack={onBack}
+      />
+      <div className='seam-main'>
+        <FormField>
+          <InputLabel>{t.nameInputLabel}</InputLabel>
+          <TextField
+            size='large'
+            clearable
+            value={name}
+            onChange={setName}
+            hasError={nameError != null}
+            helperText={nameError}
+          />
+        </FormField>
+
+        <FormField className='seam-code-field'>
+          <InputLabel>{t.codeInputLabel}</InputLabel>
+          <TextField
+            size='large'
+            clearable
+            value={code}
+            onChange={setCode}
+            onFocus={toggleCodeInputFocused}
+            onBlur={toggleCodeInputFocused}
+            hasError={codeError != null}
+          />
+          <div
+            className={classNames('seam-bottom', {
+              'seam-visible': codeInputFocused,
+            })}
+          >
+            <ul
+              className={classNames('seam-requirements', {
+                'seam-error': codeError != null,
+              })}
+            >
+              <li>{t.codeRequirementLength}</li>
+              <li>{t.codeRequirementNumbersOnly}</li>
+            </ul>
+            <Button
+              size='small'
+              onMouseDown={(e) => {
+                e.preventDefault() // Disable stealing input focus
+                generateCode()
+              }}
+            >
+              {t.codeGenerateButton}
+            </Button>
+          </div>
+        </FormField>
+        <div className='seam-actions'>
+          <Button onClick={onBack}>{t.cancel}</Button>
+          <Button variant='solid' disabled={!isFormValid} onMouseDown={save}>
```

Comment:
> Can't create a new code because of the show hide behavior on the help text.

_https://github.com/seamapi/react/pull/271#issuecomment-1631262372_

@razor-x good catch! Fixed in https://github.com/seamapi/react/pull/271/commits/ccf549f60b39e6292a86a0a50f6f90ff8376a41d

### Before

https://github.com/seamapi/react/assets/11449462/1f82879b-ecd1-43d7-a339-24c8eaa74398

_No `/create` requests sent on click`_

### After

https://github.com/seamapi/react/assets/11449462/71bb76d2-9812-4f54-8e52-ccb2359e5f2e

_Each click triggers a  `/create` request`_

Reply:
This is better, but still UX w/ the jump is not great.

Maybe the helper text should always be there? I'm thinking this is related to my comment about generating the code: we need to ensure the UI does not fustrate the user by creating or letting them create codes that work work.

This is just like those password form that only tell you your password needs to be less than 30 characters with a number and a symbol AFTER you try to use `abcdefg`.

Reply:
ohh you're talking about the UX. Might be more of a question for @dawnho, and @miken-seam then. 

Was just following spec:

![CleanShot 2023-07-12 at 10 23 52](https://github.com/seamapi/react/assets/11449462/20bc1617-f80d-4377-866d-f7c1a2863b04)

https://www.figma.com/file/Su3VO6yupz4yxe88fv0Uqa/Seam-Components?type=design&node-id=895-82543&mode=design&t=QKRKt2MSYlUSYTLb-0

Reply:
Hmm, showing it when the box is first focused is ok. Didn't realize that at first. How about as long as the box has any value we show it? That would prevent the jump.

This idea won't work if the form is supposed to be cleared on submit though 🤔 

Some open questions then for @dawnho @miken-seam 

1. What do you think of always displaying the help info for the access code box?
2. The info can be populated with the actual constraints for the device.
3. How do we handle the case where the code cannot be set by the user?
4. Can we communicate to the user they do not need to set a value for the code? Something like a checkbox next to the input "Autogenerate a code" and when the checkbox is clicked the input is disabled?

Reply:
I spoke with @dawnho in office. The consensus is we want to add the code input box and generator button, but not until we ensure the UX will handle the device code constraints. We don't want to ship a component that integrates with Seam in a way that does not use it as intended.

So, you can split out the code for this input box from this PR, and we can work on it as it's own feature.

Reply:
We should probably clear off the fields when "Save" is clicked

Reply:
> We should probably clear off the fields when "Save" is clicked

As in reset the form? or hide the Instructions / Generate code button?

Reply:
Removed the code gen in https://github.com/seamapi/react/pull/271/commits/adf6badfa2561eb010faad847cf95f463d3f290c

Also

Reply:
Removed code input in https://github.com/seamapi/react/pull/271/commits